frontend/src/app/providers/complaint-filter-provider.tsx

```diff
@@ -52,13 +52,18 @@ const mapFilters = (complaintFilters: Partial<ComplaintFilters>) => {
     natureOfComplaintFilter,
     outcomeAnimalFilter,
   } = complaintFilters;
+
+  // Parse the start and end date filters into Date objects if they exist.
+  const parsedStartDate = startDateFilter ? new Date(startDateFilter) : undefined;
+  const parsedEndDate = endDateFilter ? new Date(endDateFilter) : undefined;
```

Comment:
![image](https://github.com/user-attachments/assets/e4a157a2-6689-4a5e-af3c-40b7c5b67d6e)

The logic for the date filter is if the end date is null/empty it defaults to the current date not undefined.